### PR TITLE
Add std.string.detabber(), a range version of detab()

### PR DESCRIPTION
**Pull request by WalterBright**
Unit tests at 100%.

### CODE REVIEW

**Comment by Panke** [comment]
LGTM

File: std/string.d [hunk 1]
```diff
@@ -3076,65 +3076,148 @@ S center(S)(S s, size_t width, dchar fillChar = ' ') @trusted pure
...
+    Params:
+        s = string
+        tabSize = distance between tab stops
+
+    Returns:
+        GC allocated string with tabs replaced with spaces
   +/
-S detab(S)(S s, size_t tabSize = 8) @trusted pure
+S detab(S)(S s, size_t tabSize = 8) pure
     if (isSomeString!S)
```
Comment (JakobOvrum) [inline std/string.d]: Any reason to limit `detab` to strings unless we plan to deprecate it?

File: std/string.d [hunk 2]
```diff
@@ -3076,65 +3076,148 @@ S center(S)(S s, size_t width, dchar fillChar = ' ') @trusted pure
...
+
+    Params:
+        r = string or forward range
+        tabSize = distance between tab stops
+
+    Returns:
+        lazy forward range with tabs replaced with spaces
+  +/
+auto detabber(Range)(Range r, size_t tabSize = 8)
+    if (isForwardRange!Range)
```
Comment (JakobOvrum) [inline std/string.d]: `&& isSomeChar!(ElementEncodingType!Range)`

**Comment by kuettler** [comment]
There is no `detabber` entry in the TOC. Should there be one?

**Comment by kuettler** [comment]
Beautiful code. LGTM.

File: std/string.d (hunk 2 above)
Comment (WalterBright) [inline std/string.d]: done

**Comment by WalterBright** [comment]
> There is no  detabber  entry in the TOC. Should there be one?

added

File: std/string.d [hunk 3]
```diff
@@ -3076,65 +3077,148 @@ S center(S)(S s, size_t width, dchar fillChar = ' ') @trusted pure
 
 /++
     Replace each tab character in $(D s) with the number of spaces necessary
-    to align the following character at the next tab stop where $(D tabSize)
-    is the distance between tab stops.
+    to align the following character at the next tab stop.
+
+    Params:
+        s = string
```
Comment (andralex) [inline std/string.d]: ehm

**Comment by andralex** [comment]
nice work

**Pull request merged**